OVERVIEW.md

```diff
@@ -0,0 +1,102 @@
+# Demo service overview
+
+## Architecture
+This service is a Django based service, running with a Celery message queue system. The compiled Docker image has two entry points to run; as either the Django based web frontend, or the Celery worker. The web service runs the web interface and webhook endpoints, and the Celery worker will perform any background tasks that have been added to the message queue.
+
+This service is usable in a variety of environments but it has been configured to run in a Rancher system. The rancher configuration is a `docker-compose.yml` file at its core, with some configuration options for Rancher. The service starts with these components:
+
+- `demoservice-lb`: Load balancer to route traffic to web services
+- App runs as two services
+  - `demoservice-web`: Django frontend service
+  - `demoservice-worker`: Celery worker
+- `demoservice-db`: Postgres database
+- `demoservice-rabbit`: RabbitMQ database
+
+Separately to this, a [Traefik](https://traefik.io/) instance runs to route traffic. This will be explained a little more clearly later in the document.
+
+
+### App structure
+
+A simple overview of the folder structure:
+
+```
+├── app
+│   └── demoservice
+│       ├── libs
+│       │   └── [DJANGO/PYTHON HELPER FILES]
+│       ├── migrations
+│       │   └── [DJANGO DATABASE MIGRATION SCRIPTS]
+│       └── templates
+│           └── [DJANGO HTML TEMPLATES]
+├── bin
+│   └── [HELPER SCRIPTS]
+└── templates
+    └── [RANCHER DEPLOYMENT TEMPLATE]
+```
+
+Inside the demoservice folder there are a couple of base files:
+- `tasks.py`
+  - Celery tasks and configuration
+- `views.py`
+  - webhook endpoint
+  - frontend management pages
+
+
+## Flow
+
+### Webhooks
+- Webhook, important to check secret
+ - Listens for pull_request events of type 'opened', 'closed', and 'synchronised'.
+- ...
+- Starts ./run demo
+
+_**When receiving webhooks, it is important to always check the secret hash. ALWAYS.**_
+
+#### Github webhooks
+Github webhooks are pushed to the `/webhook/github` path. When the app receives a payload from GitHub it will check the signature against a hash made from our secret. We must check this to ensure that only webhooks from verified sources are started.
+
+Github can send a variety of event types. The type is checked against the header `HTTP_X_GITHUB_EVENT`. This then checks the type of event in the payload content. We primarily listen for `opened`, `closed`, and `synchronized`. Synchronized is triggered when a branch has been updated; often by rebasing or pushing new commits.
+
+When the webhook has been understood, it queues up a relevant task in the message queue. It will either start/update a demo or delete one. Starting a demo will stop any previously running demos, so this action can be used for updating a demo.
```

Comment:
`so this action can be used for updating a demo` why is it not implemented yet?

Reply:
@tbille that's rather mean. Presumably he didn't have time, and it's definitely out of scope of this documentation PR.

I just created the issue - https://github.com/canonical-webteam/demoservice/issues/29 - feel free to start work on it in your free time.

Reply:
I am reclarifying what I mean here